maestro/lifecycle.py

```diff
@@ -65,10 +65,7 @@ def __repr__(self):
 
     def _test(self, container=None):
         try:
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.settimeout(1)
-            s.connect((self.host, self.port))
-            s.close()
+            socket.create_connection((self.host, self.port), 1)
```

Comment:
I think you are right; close socket is needed on successful connection (https://github.com/python/cpython/blob/6f0eb93183519024cb360162bdd81b9faec97ba6/Lib/socket.py#L714). 

How about
```
s = socket.create_connection((self.host, self.port), 1)
s.close()
```